R/git-utils.R

```diff
@@ -152,26 +154,41 @@ git_branch_compare <- function(branch = git_branch_name()) {
   )
 }
 
-git_branch_push <- function(branch = git_branch_name(), force = FALSE) {
-  remote <- git_branch_tracking(branch)
-  if (is.null(remote)) {
```

Comment:
All this logic had to be pulled out into `git_branch_remote()` so I know / can guess the name of the remote associated with current branch. And I need that to look up the URL. And I need that to see if the current repo is managed with SSH or HTTPS.